pkg/utils/kubernetes/featuregates.go

```diff
@@ -0,0 +1,207 @@
+	"StartupProbe":                                   {MinVersion: "1.16", MaxVersion: "1.20"},
+	"StorageObjectInUseProtection":                   {MinVersion: "1.10", MaxVersion: "1.11"},
+	"StorageVersionAPI":                              {MinVersion: "1.20"},
+	"StorageVersionHash":                             {MinVersion: "1.14"},
+	"StreamingProxyRedirects":                        {MinVersion: "1.5", MaxVersion: "1.19"},
+	"SupportIPVSProxyMode":                           {MinVersion: "1.8", MaxVersion: "1.11"},
+	"SupportNodePidsLimit":                           {MinVersion: "1.14", MaxVersion: "1.20"},
+	"SupportPodPidsLimit":                            {MinVersion: "1.10", MaxVersion: "1.20"},
+	"SuspendJob":                                     {MinVersion: "1.21"},
+	"Sysctls":                                        {MinVersion: "1.11", MaxVersion: "1.21"},
+	"TTLAfterFinished":                               {MinVersion: "1.12"},
+	"TaintBasedEvictions":                            {MinVersion: "1.6", MaxVersion: "1.18"},
+	"TaintNodesByCondition":                          {MinVersion: "1.8", MaxVersion: "1.17"},
+	"TokenRequest":                                   {MinVersion: "1.10", MaxVersion: "1.20"},
+	"TokenRequestProjection":                         {MinVersion: "1.11", MaxVersion: "1.20"},
+	"TopologyAwareHints":                             {MinVersion: "1.21"},
+	"TopologyManager":                                {MinVersion: "1.16"},
+	"ValidateProxyRedirects":                         {MinVersion: "1.12"},
+	"VolumeCapacityPriority":                         {MinVersion: "1.21"},
+	"VolumePVCDataSource":                            {MinVersion: "1.15", MaxVersion: "1.18"},
+	"VolumeScheduling":                               {MinVersion: "1.9", MaxVersion: "1.13"},
+	"VolumeSnapshotDataSource":                       {MinVersion: "1.12", MaxVersion: "1.20"},
+	"VolumeSubpathEnvExpansion":                      {MinVersion: "1.14", MaxVersion: "1.17"},
+	"WarningHeaders":                                 {MinVersion: "1.19"},
+	"WatchBookmark":                                  {MinVersion: "1.15", MaxVersion: "1.17"},
+	"WinDSR":                                         {MinVersion: "1.14"},
+	"WinOverlay":                                     {MinVersion: "1.14"},
+	"WindowsEndpointSliceProxying":                   {MinVersion: "1.19"},
+	"WindowsGMSA":                                    {MinVersion: "1.14", MaxVersion: "1.18"},
+	"WindowsRunAsUserName":                           {MinVersion: "1.16", MaxVersion: "1.18"},
+
+	// TODO The feature gates below are listed under "Feature gates for graduated or deprecated features"
+	// but without Alpha or Beta version ranges, so it's not really clear for which version they were supported.
+	// EnableAggregatedDiscoveryTimeout	true	Deprecated	1.16	-
+	// ExecProbeTimeout	true	GA	1.20	-
+	// ExternalPolicyForExternalIP	true	GA	1.18	-
```

Comment:
`ExternalPolicyForExternalIP` was introduced in `1.18` and is planned to be removed in `1.22` (so let's not specify a max version yet?)

Reply:
This will be properly reflected when the new approach we discussed is implemented.